### PR TITLE
fix: Update Create Alert modal attributes correctly

### PR DESCRIPTION
**Pull request by olerichter00**
### Description

The selected filters (attributes) are not reflected in the "Create Alert" modal. This leads to alerts being saved without additional filters (like `medium`) and only the artist ID.

**Before:**

https://github.com/artsy/eigen/assets/4691889/e934ba85-a431-4416-a718-5f1debddd3d1

**After:**

https://github.com/artsy/eigen/assets/4691889/fed9e7e7-aee6-41b7-87a4-9bcd5090296e

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 758495699763a91891aec6e34b5f2eb2b72565ef
</p>

File: src/app/Components/Artist/ArtistArtworks/CreateSavedSearchModal.tsx [hunk 1]
```diff
@@ -59,6 +59,8 @@ export const CreateSavedSearchModal: React.FC<CreateSavedSearchModalProps> = (pr
     onComplete: handleComplete,
   }
 
+  if (!visible) return null
+
   return <CreateSavedSearchAlert visible={visible} params={params} />
```
Comment (olerichter00) [inline src/app/Components/Artist/ArtistArtworks/CreateSavedSearchModal.tsx:64]: The params get passed as initial values to the context store in `CreateSavedSearchAlert`. So they won't be updated after the first render of the component. By not rendering `CreateSavedSearchAlert` unless it's visible, the store gets newly initialized every time the modal gets visible.

**Review by nickskalkin — APPROVED** [review]
Awesome, thanks @olerichter00!

File: src/app/Components/Artist/ArtistArtworks/CreateSavedSearchModal.tsx (hunk 1 above)
Comment (MounirDhahri) [reply src/app/Components/Artist/ArtistArtworks/CreateSavedSearchModal.tsx:64]: Unfortunately this also breaks the modal closing animation as well. I would rather suggest to find a way to listen to the values and update them accordingly.

**Pull request merged**